### ISSUE TITLE
[adc_ctrl] Set ADC_EN_CTRL to read only for CSR tests.

### ISSUE THREAD

**Issue by NigelScales** [lowRISC/opentitan#10113]
The ADC_CTRL CSR tests fail because ADC_EN_CTRL is being written leading to side effects.
I've tried adding the following tags to the register and this solved the problem,

```
tags: [ // Writes to ADC_EN_CTRL will cause side-sffects
              "excl:CsrNonInitTests:CsrExclWrite"
```

**Comment by rswarbrick** [comment]
Sounds like the right thing to do. Could you open a PR?

**Comment by tjaychen** [comment]
this sounds good to me as well.

**Comment by NigelScales** [comment]
I've done a PR for just this change.
#10186

**Comment by sriyerg** [comment]
Left a comment about moving the tag to a field rather than the whole register.

**Comment by NigelScales** [comment]
Fixed by #10186
